3ds2/src/main/java/com/adyen/checkout/adyen3ds2/internal/ui/SharedChallengeStatusHandler.kt

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright (c) 2024 Adyen N.V.
+ *
+ * This file is open source and available under the MIT license. See the LICENSE file for more info.
+ *
+ * Created by oscars on 21/5/2024.
+ */
+
+package com.adyen.checkout.adyen3ds2.internal.ui
+
+import androidx.annotation.VisibleForTesting
+import com.adyen.threeds2.ChallengeResult
+import com.adyen.threeds2.ChallengeStatusHandler
+
+internal object SharedChallengeStatusHandler : ChallengeStatusHandler {
+
+    var onCompletionListener: ChallengeStatusHandler? = null
+        set(value) {
+            field = value
+            resultQueue?.let { onCompletion(it) }
+        }
+
+    private var resultQueue: ChallengeResult? = null
```

Comment:
To me this naming is a little misleading, since it is not a queue of operations.
What do you think of renaming this to `queuedResult` or keep the name the same but change the type to a list (`mutableListOf<() -> Unit>()`)?

Reply:
`queuedResult` sounds better indeed